src/plugin.ts

```diff
@@ -0,0 +1,154 @@
+            this.erc20BridgeL2 = addresses.l2Erc20DefaultBridge;
+            this.wethBridgeL1 = addresses.l1WethBridge;
+            this.wethBridgeL2 = addresses.l2WethBridge;
+        }
+        return {
+            erc20L1: this.erc20BridgeL1,
+            erc20L2: this.erc20BridgeL2,
+            wethL1: this.wethBridgeL1,
+            wethL2: this.wethBridgeL2,
+        };
+    }
+
+    async getL1Address(token: Address): Promise<string> {
+        if (token == ETH_ADDRESS) {
+            return ETH_ADDRESS;
+        } else {
+            const bridgeAddresses = await this.getDefaultBridgeAddresses();
+            const l2WethBridge = this.getL2BridgeContract(bridgeAddresses.wethL2);
+            try {
+                const l1WethToken = await l2WethBridge.methods.l1TokenAddress(token).call();
+                if (l1WethToken !== ZERO_ADDRESS) {
+                    return l1WethToken;
+                }
+            } catch (e) {
+            }
+            const erc20Bridge = this.getL2BridgeContract(bridgeAddresses.erc20L2);
+            return erc20Bridge.methods.l1TokenAddress(token).call();
+        }
+    }
+
+    async getL2Address(token: Address): Promise<string> {
+        if (token == ETH_ADDRESS) {
+            return ETH_ADDRESS;
+        } else {
+            const bridgeAddresses = await this.getDefaultBridgeAddresses();
+            const l2WethBridge = this.getL2BridgeContract(bridgeAddresses.wethL2);
+            try {
+                const l2WethToken = await l2WethBridge.methods.l2TokenAddress(token).call();
+                if (l2WethToken !== ZERO_ADDRESS) {
+                    return l2WethToken;
+                }
+            } catch (e) {
+            }
+
+            const erc20Bridge = this.getL2BridgeContract(bridgeAddresses.erc20L2)
+            return erc20Bridge.methods.l2TokenAddress(token).call();
+        }
+    }
+
+    async accountBalances(address: Address): Promise<{ [key: string]: bigint }> {
+        let balances = await this.rpc.getAllAccountBalances(address);
+        for (let token in balances) {
+            balances[token] = BigInt(balances[token]);
+        }
+        return balances;
+    }
+
+    getERC20BalanceByAddress(tokenAddress: Address, address: Address) {
+        return this.erc20(tokenAddress).methods.balanceOf(address).call()
+    }
+
+    async getErc20TokenInfo(tokenAddress: Address) {
+        const erc20Contract = this.erc20(tokenAddress)
+        const [name, symbol, decimals, totalSupply] = await Promise.all([
+            erc20Contract.methods.name().call(),
+            erc20Contract.methods.symbol().call(),
+            erc20Contract.methods.decimals().call(),
+            erc20Contract.methods.totalSupply().call(),
+        ]);
+        return {
+            name,
+            symbol,
+            decimals,
+            totalSupply
+        }
+    }
+}
+
+// Module Augmentation
+declare module "web3" {
+    interface Web3Context {
```

Comment:
Actually, no need to augment the `Web3Context`, augmenting `Web3` is enough.
```suggestion
    interface Web3 {
```